doc/modules/model_evaluation.rst

```diff
@@ -1536,6 +1536,16 @@ Here is a small example of usage of this function:::
     >>> brier_score_loss(y_true, y_prob > 0.5)
     0.0
 
+The Brier score can be used to assess how well a classifier is calibrated
+however, a lower Brier score does not always mean a better calibration.
+This is because the Brier score is a combination of calibration
+loss and refinement loss. Calibration loss is defined as the mean squared
+deviation from empirical probabilities derived from the slope of ROC segments.
+Refinement loss can be defined as the expected optimal loss as measured by
+the area under the optimal cost curve. Refinement loss can change
+independently from calibration loss, thus a lower Brier score does not
+necessarily mean a better calibrated model. Only when refinement loss remains
+the same does a lower Brier score always mean better calibration.
```

Comment:
I believe this paragraph was taken from this book chapter [Calibration of Machine Learning Models](http://dmip.webs.upv.es/papers/BFHRHandbook2010.pdf) by Antonio Bella, Cèsar Ferri,José Hernández-Orallo, and María José Ramírez-Quintana (full book: [Machine Learning: Concepts, Methodologies, Tools and Applications: Concepts](https://books.google.fr/books?id=1GWcHmCrl0QC)) and refers to a 2008 paper by Flach, Peter & Matsubara, Edson:
[On classification, ranking, and probability estimation](https://drops.dagstuhl.de/opus/volltexte/2008/1382/).

I think it's fine to keep the verbatim quote but:
- it should be formatted as such (using double quote or a sphinx directive)
- we should credit the authors, both of the quote in the book chapter and the original paper.

Alternatively we might want to rephrase but we would first all need to read the original paper to make sure we .

In particular, we need to clarify how the calibration term of the Brier score decomposition to the usual calibration error that is being worked on in #11096.

Reply:
I will make a suggestion in #11096 to add an explanation between the calibration loss implemented and the calibration loss here which uses ROC segments.

Reply:
Let's merge this PR as it is and refine later once we converge on the calibration error PR.